### PR TITLE
Fix : "Set featured image" button border flashes on focus

### LINKED ISSUE #65299: "Set featured image" button border flashes on focus
## Steps to reproduce

In a post, go to the Post tab in the sidebar and move the keyboard focus onto the "Set featured image" button.

## Expected behavior

The focus ring shows like the other components.

## Actual behavior

There is a lag between the focus ring appearing and the normal border disappearing.

## Video

https://github.com/user-attachments/assets/bce6d0de-5f66-438a-a746-2030f6429da2

### PR DESCRIPTION
**Pull request by Vrishabhsk**
## What?

- Add `box-shadow` style for `:focus:not(:disabled)` state of "Set featured image" button.

## Why?

- https://github.com/WordPress/gutenberg/issues/65299
> There is a lag between the focus ring appearing and the normal border disappearing.

Fixes #65299

## How?

- The `box-shadow` `inset` property isn't affected by the `transition delay`
- To prevent this, the style was implemented as per the steps mentioned here : [`#66007`](https://github.com/WordPress/gutenberg/issues/66007#issuecomment-2405194812)

### Original

https://github.com/user-attachments/assets/520910d1-b84c-4556-a7b3-d07daac49277

### Improved

https://github.com/user-attachments/assets/98262305-59ff-4f1e-86b9-b403d36c8899

### CODE REVIEW

**Review by t-hamano — COMMENTED** [review]
Thanks for the PR! 

I noticed that the moment the button was focused, the border changed to black.

See the slow motion video below:


https://github.com/user-attachments/assets/e2a47fac-2799-4717-adb3-a1585a5fdc30

I looked for the cause and found that [these three styles](https://github.com/WordPress/gutenberg/blob/4b6e08aac313d605dcfbc6ef0666bc75e62b5dbe/packages/editor/src/components/post-featured-image/style.scss#L44-L46) were affecting it. Since appropriate styles were already provided, it seemed safe to remove them.

Below is the slow motion video after removing these three styles:

https://github.com/user-attachments/assets/878d201c-8b04-42bf-8060-12733c82feed

File: packages/editor/src/components/post-featured-image/style.scss [hunk 1]
```diff
@@ -66,6 +66,11 @@
 
 .editor-post-featured-image__toggle {
 	box-shadow: inset 0 0 0 $border-width $gray-400;
+
+	&:focus:not(:disabled) {
+		// Allow smooth transition between focused and unfocused box-shadow states.
+		box-shadow: 0 0 0 currentColor inset, 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color, #3858e9);
```
Comment (mirka) [inline packages/editor/src/components/post-featured-image/style.scss]:
```suggestion
		box-shadow: 0 0 0 currentColor inset, 0 0 0 var(--wp-admin-border-width-focus) var(--wp-admin-theme-color);
```

We won't need fallbacks in this package 👍

**Comment by Vrishabhsk** [comment]
Thanks @t-hamano for the catch.
- Instead of removing all three styles, I instead removed only the `transition` line
- This prevents the` black border` which appeared for a split-second before the transition was complete.
- Also removing all three styles, does change the `border color` quite a bit from the `original`
- Here is the preview of that, slowed down

https://github.com/user-attachments/assets/bbf0c8a3-eb0c-4005-89bc-2ddd2da69a02

Let me know your thoughts on this. Thanks

**Comment by t-hamano** [comment]
Thanks for the update!

> Also removing all three styles, does change the `border color` quite a bit from the `original`

Does this have to do with [this box-shadow](https://github.com/WordPress/gutenberg/blob/98f3f6a5b3cccfcc8d94cefa9a4bee9750478132/packages/editor/src/components/post-featured-image/style.scss#L46)? This style is always supposed to be overridden [here](https://github.com/WordPress/gutenberg/blob/98f3f6a5b3cccfcc8d94cefa9a4bee9750478132/packages/editor/src/components/post-featured-image/style.scss#L68), so I'm wondering why it's affecting the transition 🤔

**Comment by Vrishabhsk** [comment]
Hi @t-hamano,
- As per computed styles for the `buttton`, it has the following styles
```
    border-width: 2px;
    border-style: outset;
    border-color: buttonborder;
    border-image: initial;
```
- This style is overridden by the `.components-button` selector, which sets `border: 0`
- The `transition` used is applied on for `all` CSS properties.
- Hence, it takes the `border` also into account, and it appears for a split second
- This is also why removing that `transition` prevents the `black border` from appearing at all

**Review by t-hamano — APPROVED** [review]
LGTM!

One last thing: what do you think about removing [this mixin](https://github.com/WordPress/gutenberg/pull/66092/files#diff-027457a66581220a0f7ca217f66814f5b813d5b8270e1d5e9a1cb8343294728bL45)? Since this mixin [exists in the Button component itself](https://github.com/WordPress/gutenberg/blob/0ee8ce4536a2fc57ddeef10b9ada4bf9ceb07cdd/packages/components/src/button/style.scss#L20), it seems safe to remove it.

**Comment by Vrishabhsk** [comment]
Hi @t-hamano 👋, I have removed the redundant mixin. Thanks for pointing that out.

**Pull request merged**